File: app/src/main/java/ch/epfl/sdp/game/Player.java
package ch.epfl.sdp.game;

import androidx.annotation.NonNull;

public class Player {
    private int id;
    private Faction faction;
    private Location lastKnownLocation;

    public Player(int id, Faction faction) {
        this.id = id;
        this.faction = faction;
        lastKnownLocation = null;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Faction getFaction() {
        return faction;
    }

    public void setFaction(Faction faction) {
        this.faction = faction;
    }

    public Location getLastKnownLocation() {
        return lastKnownLocation;
    }

    public void setLastKnownLocation(Location lastKnownLocation) {
        this.lastKnownLocation = lastKnownLocation;
    }

    @NonNull
    @Override
    public String toString() {
        return "Player{" +
                "id=" + id +
                ", faction=" + faction +
                ", lastKnownLocation=" + lastKnownLocation +
                '}';
    }
}

File: app/src/main/java/ch/epfl/sdp/game/data/Player.kt
package ch.epfl.sdp.game.data

import java.io.Serializable

/**
 * Class describing a Player
 * @param id Int: the player's ID
 */
open class Player(val id: Int) : Serializable {
    class WrongPlayerFormat: Exception()

    var lastKnownLocation: Location? = null

    override fun toString(): String {
        return "Player{" +
                "id=" + id +
                ", lastKnownLocation=" + lastKnownLocation +
                '}'
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Player

        if (id != other.id) return false
        if (lastKnownLocation != other.lastKnownLocation) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id
        result = 31 * result + (lastKnownLocation?.hashCode() ?: 0)
        return result
    }
}